src/views/Profile/PersonalInfo/RegionCityHandler.tsx

```diff
@@ -0,0 +1,175 @@
+        [inputIdentifier]: i18n.t(langTokens.admin.fieldCantBeEmpty),
+      });
+      setIsCitySelect(false);
+      setSelectRegions(null);
+      if (!regionForValue) {
+        setSelectCities(null);
+      }
+    } else {
+      setErrorFields({ ...errorFields, [inputIdentifier]: '' });
+      setNewAuthorValues({ ...newAuthorValues, [inputIdentifier]: value.id });
+      setRegionByCityId(value.id);
+      setIsCitySelect(true);
+    }
+  };
+
+  const inputRegionChangeHandler = (
+    _,
+    value: string | '',
+    inputIdentifier: string,
+  ) => {
+    if (!value) {
+      setErrorFields({
+        ...errorFields,
+        [inputIdentifier]: i18n.t(langTokens.admin.fieldCantBeEmpty),
+      });
+      setRegionForValue(null);
+      setSelectCities(null);
+      if (!isCitySelect) {
+        setSelectRegions(null);
+      }
+    } else {
+      const regionFromValue = regions.find((region) => region.name === value);
+      if (regionFromValue) {
+        setErrorFields({ ...errorFields, [inputIdentifier]: '' });
+        setNewAuthorValues({
+          ...newAuthorValues,
+          [inputIdentifier]: regionFromValue.id,
+        });
+        setCitiesByRegionId(regionFromValue.id);
+        setRegionForValue(regionFromValue);
+      }
+    }
+  };
+
+  return (
+    <Grid container spacing={2}>
+      <Grid item xs={6}>
+        <ErrorField
+          errorField={errorFields.regionId}
+          visitField={visitFields.region}
+        />
+        <Autocomplete
+          disablePortal
+          options={selectRegions || regions}
+          fullWidth
+          value={regionForValue}
+          getOptionLabel={(region: IRegion) => region.name}
+          getOptionSelected={(option, value) => option.id === value.id}
+          onInputChange={(_, value) =>
+            inputRegionChangeHandler(_, value, 'regionId')
+          }
+          renderInput={(params) => (
+            <TextField
+              variant="outlined"
+              className={classes.TextField}
+              required
+              // eslint-disable-next-line
+              {...params}
+              label={i18n.t(langTokens.admin.region)}
+              name="region"
+              onBlur={blurHandler}
+            />
+          )}
+        />
+      </Grid>
+      <Grid item xs={6}>
+        <ErrorField
+          errorField={errorFields.cityId}
+          visitField={visitFields.city}
+        />
+        <Autocomplete
+          disablePortal
+          options={selectCities || cities}
+          fullWidth
+          getOptionLabel={(city: ICity) => city.name}
+          getOptionSelected={(option, value) => option.id === value.id}
+          onChange={(_, value) => inputCityChangeHandler(_, value, 'cityId')}
+          renderInput={(params) => (
+            <TextField
+              variant="outlined"
+              className={classes.TextField}
+              required
+              // eslint-disable-next-line
+              {...params}
```

Comment:
the same, `{...params}` in the middle better to be avoided

Reply:
moved to the very end